File: tests/PlaywrightNode.Tests/pages/OrdersPage.js
class OrdersPage {
    constructor(page) {
        this.page = page;
    }

    async goto() {
        await this.page.goto('/Orders');
    }

    async isOrderVisible(productName) {
        const orderItem = this.page.locator('.order-item').filter({ hasText: productName }).first();
        return await orderItem.isVisible();
    }

    async getEmptyMessage() {
        return await this.page.locator("p:has-text('You haven\\'t placed any orders yet.')").textContent();
    }
}

module.exports = { OrdersPage };

File: tests/E2E.Tests/Pages/OrdersPage.cs
using Microsoft.Playwright;

namespace E2E.Tests.Pages;

public class OrdersPage
{
    private readonly IPage _page;

    public OrdersPage(IPage page) => _page = page;

    public async Task GotoAsync(string baseUrl) => await _page.GotoAsync($"{baseUrl}/Orders");

    public async Task<bool> IsOrderVisibleAsync(string productName)
    {
        var orderItem = _page.Locator(".order-item").Filter(new() { HasText = productName }).First;
        return await orderItem.IsVisibleAsync();
    }

    public async Task<string?> GetEmptyMessageAsync()
    {
        return await _page.Locator("p:has-text('You haven\\'t placed any orders yet.')").TextContentAsync();
    }
}
